homeassistant/components/control4/__init__.py

```diff
@@ -0,0 +1,226 @@
+"""The Control4 integration."""
+import asyncio
+import json
+import logging
+
+from aiohttp import client_exceptions
+from pyControl4.account import C4Account
+from pyControl4.director import C4Director
+from pyControl4.error_handling import BadCredentials
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_TOKEN,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client, device_registry as dr, entity
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    CONF_ACCOUNT,
+    CONF_CONFIG_LISTENER,
+    CONF_CONTROLLER_UNIQUE_ID,
+    CONF_DIRECTOR,
+    CONF_DIRECTOR_ALL_ITEMS,
+    CONF_DIRECTOR_MODEL,
+    CONF_DIRECTOR_SW_VERSION,
+    CONF_DIRECTOR_TOKEN_EXPIRATION,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["light"]
+
+
+async def async_setup(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Stub to allow setting up this component.
+
+    Configuration through YAML is not supported at this time.
+    """
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Control4 from a config entry."""
+    entry_data = hass.data[DOMAIN].setdefault(entry.entry_id, {})
+    account_session = aiohttp_client.async_get_clientsession(hass)
+
+    config = entry.data
+    account = C4Account(config[CONF_USERNAME], config[CONF_PASSWORD], account_session)
+    try:
+        await account.getAccountBearerToken()
+    except client_exceptions.ClientError as exception:
+        _LOGGER.error("Error connecting to Control4 account API: %s", exception)
+        raise ConfigEntryNotReady
+    except BadCredentials as exception:
+        _LOGGER.error(
+            "Error authenticating with Control4 account API, incorrect username or password: %s",
+            exception,
+        )
+        raise ConfigEntryNotReady
```

Comment:
This should not be a `ConfigEntryNotReady` because we can't recover from this. For this PR, let's just `return False`. In a future PR you can start a new config flow to gather the necessary info to fix this config entry.

Reply:
```suggestion
        return False
```